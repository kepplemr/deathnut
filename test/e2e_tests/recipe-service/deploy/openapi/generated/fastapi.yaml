components:
  schemas:
    DeathnutAuthSchema:
      properties:
        grants:
          items:
            type: string
          title: Grants
          type: array
        id:
          title: Id
          type: string
        requires:
          title: Requires
          type: string
        revoke:
          default: false
          title: Revoke
          type: boolean
        user:
          title: User
          type: string
      required:
      - id
      - user
      - requires
      - grants
      title: DeathnutAuthSchema
      type: object
    HTTPValidationError:
      properties:
        detail:
          items:
            $ref: '#/components/schemas/ValidationError'
          title: Detail
          type: array
      title: HTTPValidationError
      type: object
    PartialRecipe:
      properties:
        ingredients:
          items:
            type: string
          title: Ingredients
          type: array
        title:
          title: Title
          type: string
      title: PartialRecipe
      type: object
    Recipe:
      properties:
        ingredients:
          items:
            type: string
          title: Ingredients
          type: array
        title:
          title: Title
          type: string
      required:
      - title
      - ingredients
      title: Recipe
      type: object
    RecipeWithId:
      properties:
        id:
          title: Id
          type: string
        ingredients:
          items:
            type: string
          title: Ingredients
          type: array
        title:
          title: Title
          type: string
      required:
      - title
      - ingredients
      - id
      title: RecipeWithId
      type: object
    ValidationError:
      properties:
        loc:
          items:
            type: string
          title: Location
          type: array
        msg:
          title: Message
          type: string
        type:
          title: Error Type
          type: string
      required:
      - loc
      - msg
      - type
      title: ValidationError
      type: object
info:
  title: FastAPI
  version: 0.1.0
openapi: 3.0.2
paths:
  /auth-recipe:
    post:
      operationId: auth_auth_recipe_post
      requestBody:
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/DeathnutAuthSchema'
        required: true
      responses:
        '200':
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/DeathnutAuthSchema'
          description: Successful Response
        '422':
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/HTTPValidationError'
          description: Validation Error
      summary: Auth
  /recipe:
    post:
      operationId: create_recipe_recipe_post
      requestBody:
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/Recipe'
        required: true
      responses:
        '200':
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/RecipeWithId'
          description: Successful Response
        '422':
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/HTTPValidationError'
          description: Validation Error
      summary: Create Recipe
  /recipe/{id}:
    get:
      operationId: post_recipe_recipe__id__get
      parameters:
      - in: path
        name: id
        required: true
        schema:
          title: Id
          type: string
      responses:
        '200':
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/RecipeWithId'
          description: Successful Response
        '422':
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/HTTPValidationError'
          description: Validation Error
      summary: Post Recipe
    patch:
      operationId: patch_recipe_recipe__id__patch
      parameters:
      - in: path
        name: id
        required: true
        schema:
          title: Id
          type: string
      requestBody:
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/PartialRecipe'
        required: true
      responses:
        '200':
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/RecipeWithId'
          description: Successful Response
        '422':
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/HTTPValidationError'
          description: Validation Error
      summary: Patch Recipe
schemes:
- http
- https
security:
- api_key: []
securityDefinitions:
  api_key:
    in: query
    name: key
    type: apiKey
  firebase-whatever-production:
    authorizationUrl: ''
    flow: implicit
    type: oauth2
    x-google-audiences: recipe-service
    x-google-issuer: jwt-test@wellio-dev-michael.iam.gserviceaccount.com
    x-google-jwks_uri: https://www.googleapis.com/robot/v1/metadata/x509/jwt-test@wellio-dev-michael.iam.gserviceaccount.com
