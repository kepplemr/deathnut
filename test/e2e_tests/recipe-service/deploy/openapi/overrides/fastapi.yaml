filename: fastapi.yaml
host: recipe-service-fastapi.endpoints.wellio-dev-michael.cloud.goog
info:
  description: Example recipe service
  title: Recipe Service (FastApi)
security:
  - firebase-whatever-production: []
