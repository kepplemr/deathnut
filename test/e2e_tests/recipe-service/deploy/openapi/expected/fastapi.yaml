definitions:
  DeathnutAuthSchema:
    properties:
      grants:
        items:
          type: string
        title: Grants
        type: array
      id:
        title: Id
        type: string
      requires:
        title: Requires
        type: string
      revoke:
        default: false
        title: Revoke
        type: boolean
      user:
        title: User
        type: string
    required:
    - id
    - user
    - requires
    - grants
    title: DeathnutAuthSchema
    type: object
  HTTPValidationError:
    properties:
      detail:
        items:
          $ref: '#/definitions/ValidationError'
        title: Detail
        type: array
    title: HTTPValidationError
    type: object
  PartialRecipe:
    properties:
      ingredients:
        items:
          type: string
        title: Ingredients
        type: array
      title:
        title: Title
        type: string
    title: PartialRecipe
    type: object
  Recipe:
    properties:
      ingredients:
        items:
          type: string
        title: Ingredients
        type: array
      title:
        title: Title
        type: string
    required:
    - title
    - ingredients
    title: Recipe
    type: object
  RecipeWithId:
    properties:
      id:
        title: Id
        type: string
      ingredients:
        items:
          type: string
        title: Ingredients
        type: array
      title:
        title: Title
        type: string
    required:
    - title
    - ingredients
    - id
    title: RecipeWithId
    type: object
  ValidationError:
    properties:
      loc:
        items:
          type: string
        title: Location
        type: array
      msg:
        title: Message
        type: string
      type:
        title: Error Type
        type: string
    required:
    - loc
    - msg
    - type
    title: ValidationError
    type: object
host: recipe-service-fastapi.endpoints.wellio-dev-michael.cloud.goog
info:
  description: Example recipe service
  title: Recipe Service (FastApi)
  version: 0.1.0
paths:
  /auth-recipe:
    post:
      consumes:
      - application/json
      operationId: auth_auth_recipe_post
      parameters:
      - in: body
        name: body
        required: true
        schema:
          $ref: '#/definitions/DeathnutAuthSchema'
      produces:
      - application/json
      responses:
        '200':
          description: Successful Response
          schema:
            $ref: '#/definitions/DeathnutAuthSchema'
        '422':
          description: Validation Error
          schema:
            $ref: '#/definitions/HTTPValidationError'
      summary: Auth
  /recipe:
    get:
      operationId: get_recipes_recipe_get
      parameters:
      - default: 10
        in: query
        name: limit
        required: false
        type: integer
      produces:
      - application/json
      responses:
        '200':
          description: Successful Response
          schema:
            items:
              $ref: '#/definitions/RecipeWithId'
            title: Response Get Recipes Recipe Get
            type: array
        '422':
          description: Validation Error
          schema:
            $ref: '#/definitions/HTTPValidationError'
      summary: Get Recipes
    post:
      consumes:
      - application/json
      operationId: create_recipe_recipe_post
      parameters:
      - in: body
        name: body
        required: true
        schema:
          $ref: '#/definitions/Recipe'
      produces:
      - application/json
      responses:
        '200':
          description: Successful Response
          schema:
            $ref: '#/definitions/RecipeWithId'
        '422':
          description: Validation Error
          schema:
            $ref: '#/definitions/HTTPValidationError'
      summary: Create Recipe
  /recipe/{id}:
    get:
      operationId: get_recipe_recipe__id__get
      parameters:
      - in: path
        name: id
        required: true
        type: string
      produces:
      - application/json
      responses:
        '200':
          description: Successful Response
          schema:
            $ref: '#/definitions/RecipeWithId'
        '422':
          description: Validation Error
          schema:
            $ref: '#/definitions/HTTPValidationError'
      summary: Get Recipe
    patch:
      consumes:
      - application/json
      operationId: patch_recipe_recipe__id__patch
      parameters:
      - in: path
        name: id
        required: true
        type: string
      - in: body
        name: body
        required: true
        schema:
          $ref: '#/definitions/PartialRecipe'
      produces:
      - application/json
      responses:
        '200':
          description: Successful Response
          schema:
            $ref: '#/definitions/RecipeWithId'
        '422':
          description: Validation Error
          schema:
            $ref: '#/definitions/HTTPValidationError'
      summary: Patch Recipe
schemes:
- http
- https
security:
- firebase-whatever-production: []
securityDefinitions:
  api_key:
    in: query
    name: key
    type: apiKey
  firebase-whatever-production:
    authorizationUrl: ''
    flow: implicit
    type: oauth2
    x-google-audiences: recipe-service
    x-google-issuer: jwt-test@wellio-dev-michael.iam.gserviceaccount.com
    x-google-jwks_uri: https://www.googleapis.com/robot/v1/metadata/x509/jwt-test@wellio-dev-michael.iam.gserviceaccount.com
swagger: '2.0'
x-components: {}
